src/http/multipart.rs

```diff
@@ -20,27 +20,17 @@ pub struct Multipart<'a> {
 }
 
 impl<'a> Multipart<'a> {
-    pub(crate) async fn build_form(&mut self, client: &Client) -> Result<Form> {
+    pub(crate) async fn build_form(self, client: &Client) -> Result<Form> {
         let mut multipart = Form::new();
 
-        for (file_num, file) in self.files.iter_mut().enumerate() {
+        for (file_num, file) in self.files.into_iter().enumerate() {
             // For endpoints that require a single file (e.g. create sticker),
             // it will error if the part name is not `file`.
             // https://github.com/discord/discord-api-docs/issues/2064#issuecomment-691650970
             let part_name =
                 if file_num == 0 { "file".to_string() } else { format!("file{}", file_num) };
 
-            let data = file.data(client).await?;
-            let filename = file.filename()?;
-
-            // Modify current AttachmentType to Bytes variant to prevent the
-            // need for another disk read or network request when retrying
-            if let AttachmentType::Path(_) | AttachmentType::Image(_) = file {
-                *file = AttachmentType::Bytes {
-                    data: data.clone().into(),
-                    filename: filename.clone().unwrap_or_default(),
-                };
-            }
```

Comment:
As far as I understand it, nowhere in the lib is a multipart actually re-used, since everything takes `Request`s by ownership instead of borrowing?

Reply:
Yes